CRM/MembershipExtras/Page/EditContributionRecurLineItems.php

```diff
@@ -278,17 +280,47 @@ private function setNextLineItemMembershipTypes() {
    * @throws \Exception
    */
   private function calculateCurrentPeriodStartDate() {
-    return $this->getEarliestMembershipStartDate($this->currentPeriodLineItems);
+    return $this->getEarliestLineStartDate($this->currentPeriodLineItems);
   }
 
   /**
    * Calculates current period end date.
    *
    * @return string
    *   End date of current period.
+   *
+   * @throws \Exception
    */
   private function calculateCurrentPeriodEndDate() {
-    return $this->getLargestMembershipEndDate($this->currentPeriodLineItems);
+    if (!isset($this->contribRecur['end_date'])) {
+      return $this->getLargestMembershipEndDate($this->currentPeriodLineItems);
+    }
+
+    $numberOfInstallments = 1;
+    if (!empty($this->contribRecur['installments'])) {
+      $numberOfInstallments = $this->contribRecur['installments'];
+    }
+
+    $intervalLength = CRM_Utils_Array::value('frequency_interval', $this->contribRecur, 0) * $numberOfInstallments;
```

Comment:
but this feels different to me from what we are trying to achieve, start and end date calculations should be only based on the membership dates according to our discussion and  it is not necessarily will match the recur contribution dates and interval

Reply:
The thing is, after renwal, the original membership end_date will be gone for ever. For example, we acquire a membership from 2018-01-01 to 2018-12-31. We complete all payments and renew it. Now the membership end date will be 2019-12-31. We cannot show that end date when we look at the previous period, as the end date of that period is and will always be 2018-12-31. 

Bear in mind the end date we're recording on both the recurring contribution and the recurring lines is the last recieve date of a payement plan. So I thought it better to use the original start date of the period to calculate the original end date. We could also look at the current end date and subtract as many terms as necessary, but I thought that could be even harder, as we'd have to find out what term the user is viewing. Say a user has had an annual membership renew for 10 years, and we open the manage installments form for the 7th term... we'd have to cycle through all the terms until that one (7th) to know we're looking at the seventh, and use the current end date to calculate the end date of the seventh term...

Reply:
I've removed this logic, as we no longer need it, since we are not allowing to View/Modify Future Installments for prior recurring contributions in a payment plan.